### ISSUE TITLE
AzureBlobComputeLogManager fails on missing secret_key

### ISSUE THREAD

**Issue by HynekBlaha** [dagster-io/dagster#23696]
### Dagster version

1.8.1

### What's the issue?

When I run dagster dev locally with `deployment.yaml`
```
compute_logs:
  module: dagster_azure.blob.compute_log_manager
  class: AzureBlobComputeLogManager
  config:
    storage_account: my_storage_account
    container: dagster
    default_azure_credential: {}
```
Everything works fine.

When I run it in kubernetes configured as:
```
computeLogManager:
  type: AzureBlobComputeLogManager
  config:
    azureBlobComputeLogManager:
      storageAccount: my_storage_account
      container: dagster
      defaultAzureCredential: {}
```
I get this error:
```
dagster._core.errors.DagsterInvalidConfigError: Errors whilst loading configuration for {'storage_account': <dagster._config.source.StringSourceType object at 0x7f1b8c166140>, 'container': <dagster._config.source.StringSourceType object at 0x7f1b8c166140>, 'secret_key': Field(<dagster._config.source.StringSourceType object at 0x7f1b8c166140>, default=@, is_required=False), 'default_azure_credential': Field(<dagster._config.config_type.Noneable object at 0x7f1b6a262b00>, default=None, is_required=False), 'local_dir': Field(<dagster._config.source.StringSourceType object at 0x7f1b8c166140>, default=@, is_required=False), 'prefix': Field(<dagster._config.source.StringSourceType object at 0x7f1b8c166140>, default=dagster, is_required=False), 'upload_interval': Field(<dagster._config.config_type.Noneable object at 0x7f1b6a260d90>, default=None, is_required=False)}.
    Error 1: Value at path root:secret_key must not be None. Expected "(String | { env: String })"

  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/api.py", line 760, in job_execution_iterator
    for event in job_context.executor.execute(job_context, execution_plan):
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/executor/in_process.py", line 55, in execute
    yield from iter(
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/api.py", line 874, in __iter__
    yield from self.iterator(
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/executor/in_process.py", line 26, in inprocess_execution_iterator
    yield from inner_plan_execution_iterator(
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/plan/execute_plan.py", line 39, in inner_plan_execution_iterator
    compute_log_manager = job_context.instance.compute_log_manager
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/instance/__init__.py", line 817, in compute_log_manager
    compute_log_manager = cast(InstanceRef, self._ref).compute_log_manager
  File "/usr/local/lib/python3.10/site-packages/dagster/_core/instance/ref.py", line 533, in compute_log_manager
    return self.compute_logs_data.rehydrate(as_type=ComputeLogManager)
  File "/usr/local/lib/python3.10/site-packages/dagster/_serdes/config_class.py", line 124, in rehydrate
    raise DagsterInvalidConfigError(
```

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by HynekBlaha** [comment]
Since [this test](https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-azure/dagster_azure_tests/blob_tests/test_compute_log_manager.py#L376) is passed correctly, it lead me to dive into helm chart. There I found, that the `secret_key` is templated even if it's null.

When I replace this line: https://github.com/dagster-io/dagster/blob/f8832fc966611ad6d56857532ed9fc590fdccb61/helm/dagster/templates/helpers/instance/_compute-log-manager.tpl#L23

with: 
```
  {{- if $azureBlobComputeLogManagerConfig.secretKey }}
  secret_key: {{ include "stringSource" $azureBlobComputeLogManagerConfig.secretKey }}
  {{- end }}
```
,then the errors disappear and I am able to store and fetch the logs for web UI :)

**Comment by HynekBlaha** [comment]
Why did you close this issue @cmpadden?

**Comment by HynekBlaha** [comment]
Fixed in: #23700
